overlays/haskell.nix

```diff
@@ -555,25 +555,30 @@ final: prev: {
                       });
                     }
                 ) rawProject.hsPkgs
-                // {
-                  # These are functions not packages
-                  inherit (rawProject.hsPkgs) shellFor makeConfigFiles ghcWithHoogle ghcWithPackages;
-              });
+                // (final.lib.mapAttrs (n:
+                      __trace "project.hsPkgs.${n} has been deprecated in favour of project.${n}"
+                    ) {
+                      # These are functions not packages
+                      inherit (rawProject.hsPkgs) shellFor makeConfigFiles ghcWithHoogle ghcWithPackages;
+                    }
+                  )
+              );
 
             projectCoverageReport = haskellLib.projectCoverageReport (map (pkg: pkg.coverageReport) (final.lib.attrValues (haskellLib.selectProjectPackages hsPkgs)));
+
+            inherit (rawProject.hsPkgs) makeConfigFiles ghcWithHoogle ghcWithPackages shellFor;
+
+            # Provide `nix-shell -A shells.ghc` for users migrating from the reflex-platform.
+            # But we should encourage use of `nix-shell -A shellFor`
+            shells.ghc = project.shellFor {};
```

Comment:
I'm not sure about this ... is it necessary? If so, then where is `shells.ghcjs`, if you get what I mean?

Reply:
This bit is not new in this PR (just moving it from the functions below).  I found this `shells.ghc` useful as I built `reflex-platform` support into Leksah and it expects that attribute to exist.  For cross compiling Leksah also supports `shells.ghcjs`, but I'm keen to drop it though in favour of a single `shellFor` with cross compilers (since that will work with vscode's nix tooling too).

Reply:
Hm, at least if we're going to encourage people not to use it let's emit a message.